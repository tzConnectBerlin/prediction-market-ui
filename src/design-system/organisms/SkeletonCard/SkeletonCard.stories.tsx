import * as React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';
import { SkeletonCard, SkeletonCardProps } from './SkeletonCard';

export default {
  title: 'Organism/SkeletonCard',
  component: SkeletonCard,
} as Meta;

const Template: Story<SkeletonCardProps> = (args) => <SkeletonCard {...args} />;

export const DefaultCard = Template.bind({});

export const AltCard = Template.bind({});
AltCard.args = {
  labelList: ['hello', 'testing', 'card'],
};
