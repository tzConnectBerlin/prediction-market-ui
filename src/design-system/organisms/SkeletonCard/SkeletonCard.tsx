import * as React from 'react';
import styled from '@emotion/styled';
import { Card, CardHeader, Grid, Skeleton, useMediaQuery, useTheme } from '@mui/material';
import { useTranslation } from 'react-i18next';
import { Typography, TypographyProps } from '../../atoms/Typography';
import { TokenType } from '../../../interfaces';

interface StyledLabelProps {
  icon?: string;
  fontColor?: string;
}

const StyledGrid = styled(Grid)`
  font-size: 0.8em;
  padding: 1em;
  margin-top: auto;
`;

const StyledLabel = styled.div<StyledLabelProps>`
  color: ${({ fontColor }) => fontColor};
  padding: 0.2em;
  &.hasIcon {
    display: inline-flex;
    flex-direction: row;
    align-items: center;
    & > * {
      margin-left: 0.2em;
    }
  }
`;

const StyledCard = styled(Card)`
  margin: 1em;
  width: 20em;
  cursor: pointer;
  display: flex;
  flex-direction: column;
`;
const StyledTitle = styled.div`
  display: -webkit-box;
  -webkit-line-clamp: 3;
  -webkit-box-orient: vertical;
  overflow: hidden;
  padding: 0.3em 0;
`;

export interface SkeletonCardProps {
  onClick?: () => void | Promise<void>;
  labelList?: (TokenType | string)[];
  titleSize?: TypographyProps['size'];
}

const defaultTitleSize = 'h2';

export const SkeletonCard: React.FC<SkeletonCardProps> = ({
  labelList,
  onClick,
  titleSize = defaultTitleSize,
}) => {
  const theme = useTheme();
  const { t } = useTranslation('common');
  const isMobile = useMediaQuery(theme.breakpoints.down('sm'));
  const innerLabelList = React.useMemo(() => {
    if (labelList) {
      return labelList;
    }
    return [t(TokenType.yes), t(TokenType.no), t('weekly'), t('liquidity')];
  }, [labelList, t]);
  return (
    <StyledCard onClick={onClick}>
      <CardHeader
        avatar={<Skeleton variant="circular" width={40} height={40} />}
        title={
          <>
            <Grid
              container
              direction="row"
              spacing={1}
              justifyContent={isMobile ? 'center' : 'inherit'}
            >
              <Grid item flexGrow={1}>
                <Skeleton />
              </Grid>
            </Grid>
            <StyledTitle>
              <Typography size={titleSize} fontWeight="bold">
                <Skeleton />
              </Typography>
            </StyledTitle>
          </>
        }
      />

      <StyledGrid container spacing={1}>
        <Grid container item xs={12} spacing={2}>
          {innerLabelList.map((item, i) => (
            <Grid item xs={6} key={i}>
              <StyledLabel fontColor={theme.palette.text.secondary}>
                <Typography size="h4">{item}</Typography>
              </StyledLabel>
              <StyledLabel>
                <Typography size="h3">
                  <Skeleton />
                </Typography>
              </StyledLabel>
            </Grid>
          ))}
        </Grid>
      </StyledGrid>
    </StyledCard>
  );
};
