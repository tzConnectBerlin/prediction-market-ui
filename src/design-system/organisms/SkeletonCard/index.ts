export { SkeletonCard, SkeletonCard as default } from './SkeletonCard';
export type { SkeletonCardProps } from './SkeletonCard';
