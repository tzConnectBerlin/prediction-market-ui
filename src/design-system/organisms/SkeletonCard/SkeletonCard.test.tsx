import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { SkeletonCard } from './SkeletonCard';

describe('Snapshot testing SkeletonCard Component', () => {
  it('renders correctly', () => {
    const Card = renderer.create(<SkeletonCard />).toJSON();
    expect(Card).toMatchSnapshot();
  });

  it('renders different labels', async () => {
    const { getByText } = render(<SkeletonCard labelList={['WEEKLY', 'VOLUME']} />);

    expect(getByText(/WEEKLY/i)).toBeInTheDocument();
    expect(getByText(/VOLUME/i)).toBeInTheDocument();
  });
});
