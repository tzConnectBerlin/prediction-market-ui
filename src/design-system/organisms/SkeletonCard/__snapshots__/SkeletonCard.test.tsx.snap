// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing SkeletonCard Component renders correctly 1`] = `
@keyframes animation-0 {
  0% {
    opacity: 1;
  }

  50% {
    opacity: 0.4;
  }

  100% {
    opacity: 1;
  }
}

@keyframes animation-0 {
  0% {
    opacity: 1;
  }

  50% {
    opacity: 0.4;
  }

  100% {
    opacity: 1;
  }
}

.emotion-0 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  overflow: hidden;
  margin: 1em;
  width: 20em;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 16px;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  margin-right: 16px;
}

.emotion-3 {
  display: block;
  background-color: rgba(0, 0, 0, 0.11);
  height: 1.2em;
  border-radius: 50%;
  -webkit-animation: animation-0 1.5s ease-in-out 0.5s infinite;
  animation: animation-0 1.5s ease-in-out 0.5s infinite;
}

.emotion-4 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.emotion-5 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
  display: block;
}

.emotion-6 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-top: -8px;
  width: calc(100% + 8px);
  margin-left: -8px;
  -webkit-box-pack: inherit;
  -ms-flex-pack: inherit;
  -webkit-justify-content: inherit;
  justify-content: inherit;
}

.emotion-6>.MuiGrid-item {
  padding-top: 8px;
}

.emotion-6>.MuiGrid-item {
  padding-left: 8px;
}

.emotion-7 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.emotion-8 {
  display: block;
  background-color: rgba(0, 0, 0, 0.11);
  height: auto;
  margin-top: 0;
  margin-bottom: 0;
  transform-origin: 0 55%;
  -webkit-transform: scale(1, 0.60);
  -moz-transform: scale(1, 0.60);
  -ms-transform: scale(1, 0.60);
  transform: scale(1, 0.60);
  border-radius: 4px/6.7px;
  -webkit-animation: animation-0 1.5s ease-in-out 0.5s infinite;
  animation: animation-0 1.5s ease-in-out 0.5s infinite;
}

.emotion-8:empty:before {
  content: "\\00a0";
}

.emotion-9 {
  display: -webkit-box;
  -webkit-line-clamp: 3;
  -webkit-box-orient: vertical;
  overflow: hidden;
  padding: 0.3em 0;
}

.emotion-10 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 300;
  font-size: 3.75rem;
  line-height: 1.2;
  letter-spacing: -0.00833em;
  font-weight: 700;
}

.emotion-10.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
}

.emotion-12 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-top: -8px;
  width: calc(100% + 8px);
  margin-left: -8px;
  font-size: 0.8em;
  padding: 1em;
  margin-top: auto;
}

.emotion-12>.MuiGrid-item {
  padding-top: 8px;
}

.emotion-12>.MuiGrid-item {
  padding-left: 8px;
}

.emotion-13 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-top: -16px;
  width: calc(100% + 16px);
  margin-left: -16px;
  -webkit-flex-basis: calc(100% + 16px);
  -ms-flex-preferred-size: calc(100% + 16px);
  flex-basis: calc(100% + 16px);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: calc(100% + 16px);
}

.emotion-13>.MuiGrid-item {
  padding-top: 16px;
}

.emotion-13>.MuiGrid-item {
  padding-left: 16px;
}

.emotion-14 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-basis: 50%;
  -ms-flex-preferred-size: 50%;
  flex-basis: 50%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 50%;
}

.emotion-15 {
  color: rgba(0, 0, 0, 0.6);
  padding: 0.2em;
}

.emotion-15.hasIcon {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-15.hasIcon>* {
  margin-left: 0.2em;
}

.emotion-16 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 2.125rem;
  line-height: 1.235;
  letter-spacing: 0.00735em;
}

.emotion-16.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
}

.emotion-17 {
  padding: 0.2em;
}

.emotion-17.hasIcon {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-17.hasIcon>* {
  margin-left: 0.2em;
}

.emotion-18 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 3rem;
  line-height: 1.167;
  letter-spacing: 0em;
}

.emotion-18.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
}

<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root emotion-0"
>
  <div
    className="MuiCardHeader-root emotion-1"
  >
    <div
      className="MuiCardHeader-avatar emotion-2"
    >
      <span
        className="MuiSkeleton-root MuiSkeleton-circular MuiSkeleton-pulse emotion-3"
        style={
          Object {
            "height": 40,
            "width": 40,
          }
        }
      />
    </div>
    <div
      className="MuiCardHeader-content emotion-4"
    >
      <span
        className="MuiTypography-root MuiTypography-body2 MuiCardHeader-title emotion-5"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 emotion-6"
        >
          <div
            className="MuiGrid-root MuiGrid-item emotion-7"
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse emotion-8"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </div>
        </div>
        <div
          className="emotion-9"
        >
          <h2
            className="MuiTypography-root MuiTypography-h2  emotion-10"
            lines={3}
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse emotion-8"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </h2>
        </div>
      </span>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 emotion-12"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-2 MuiGrid-grid-xs-12 emotion-13"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 emotion-14"
      >
        <div
          className="emotion-15"
        >
          <h4
            className="MuiTypography-root MuiTypography-h4  emotion-16"
            lines={3}
          >
            Yes
          </h4>
        </div>
        <div
          className="emotion-17"
        >
          <h3
            className="MuiTypography-root MuiTypography-h3  emotion-18"
            lines={3}
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse emotion-8"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </h3>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 emotion-14"
      >
        <div
          className="emotion-15"
        >
          <h4
            className="MuiTypography-root MuiTypography-h4  emotion-16"
            lines={3}
          >
            No
          </h4>
        </div>
        <div
          className="emotion-17"
        >
          <h3
            className="MuiTypography-root MuiTypography-h3  emotion-18"
            lines={3}
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse emotion-8"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </h3>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 emotion-14"
      >
        <div
          className="emotion-15"
        >
          <h4
            className="MuiTypography-root MuiTypography-h4  emotion-16"
            lines={3}
          >
            weekly
          </h4>
        </div>
        <div
          className="emotion-17"
        >
          <h3
            className="MuiTypography-root MuiTypography-h3  emotion-18"
            lines={3}
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse emotion-8"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </h3>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 emotion-14"
      >
        <div
          className="emotion-15"
        >
          <h4
            className="MuiTypography-root MuiTypography-h4  emotion-16"
            lines={3}
          >
            liquidity
          </h4>
        </div>
        <div
          className="emotion-17"
        >
          <h3
            className="MuiTypography-root MuiTypography-h3  emotion-18"
            lines={3}
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse emotion-8"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </h3>
        </div>
      </div>
    </div>
  </div>
</div>
`;
